src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -85,69 +135,63 @@ def get(self, request: Request, project) -> Response:
+            scope.set_tag("organization.slug", project.organization.slug)
             for config in configs:
-                if not filepath.startswith(config.stack_root) and not frame:
-                    last_error = "stack_root_mismatch"
+                # The code mapping does not match and we're not dealing with a mobile platform
+                if not filepath.startswith(config.stack_root) and not mobile_frame:
                     continue
 
-                result["config"] = serialize(config, request.user)
+                result = get_link(config, filepath, ctx["commit_id"])
+
+                # For mobile we try a second time by munging the file path
+                if not result["sourceUrl"] and mobile_frame:
+                    result = try_path_munging(filepath, mobile_frame, ctx)
+
+                current_config = {
+                    "config": serialize(config, request.user),
+                    "result": result,
+                }
+
                 # use the provider key to be able to split up stacktrace
                 # link metrics by integration type
                 provider = result["config"]["provider"]["key"]
+                # print("ARMEN")
+                # print(provider)
                 scope.set_tag("integration_provider", provider)
-                scope.set_tag("stacktrace_link.platform", platform)
-
-                link, attempted_url, get_link_error = get_link(
-                    config, filepath, config.default_branch, commit_id
-                )
-
-                if not link and frame:
-                    frame["filename"] = filepath
-                    munged_frames = munged_filename_and_frames(
-                        platform, [frame], "munged_filename", sdk_name=sdk_name
-                    )
-                    if munged_frames:
-                        munged_frame: Mapping[str, Any] = munged_frames[1][0]
-                        munged_filename = str(munged_frame.get("munged_filename"))
-                        if munged_filename:
-                            if not filepath.startswith(
-                                config.stack_root
-                            ) and not munged_filename.startswith(config.stack_root):
-                                last_error = "stack_root_mismatch"
-                                continue
-
-                            link, attempted_url, error = get_link(
-                                config, munged_filename, config.default_branch, commit_id
-                            )
 
                 # it's possible for the link to be None, and in that
                 # case it means we could not find a match for the
                 # configuration
-                result["sourceUrl"] = link
-                if not link:
-                    last_error = get_link_error
-                    result["attemptedUrl"] = attempted_url
+                if not result["sourceUrl"]:
+                    matched_code_mappings.append(current_config)
```

Comment:
This will create a list of all code mappings that were attempted and its result.